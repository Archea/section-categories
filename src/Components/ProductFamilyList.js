import React, { Component } from 'react'
import ProductFamily from './ProductFamily'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'

export class ProductFamilyList extends Component {
  render() {
    if (this.props.getAll && this.props.getAll.loading) {
      return <div>Loading</div>
    }

    if (this.props.getAll && this.props.getAll.error) {
      console.log('GetAll ProductFamily error:' + this.props.getAll.error)
      return <div>Error</div>
    }
    //this.props.match.params.name
    const ProductFamilysToRender = this.props.getAll.productFamilyCategory
      .associatedFamilies
    return (
      <div className="ProductFamilyList">
        <h3>{this.props.getAll.productFamilyCategory.name}</h3>
        {ProductFamilysToRender.map(productFamily => (
          <ProductFamily key={productFamily.id} productFamily={productFamily} />
        ))}
      </div>
    )
  }
}

const PRODUCTFAMILIES_QUERY = gql`
  query GetPFs($categoryId: String!) {
    productFamilyCategory(id: $categoryId) {
      name
      associatedFamilies {
        id
        familyId
        name
      }
    }
  }
`
export default graphql(PRODUCTFAMILIES_QUERY, { name: 'getAll' })(
  ProductFamilyList
)
