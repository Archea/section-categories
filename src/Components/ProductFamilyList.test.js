import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, configure } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { ProductFamilyList } from './ProductFamilyList'

configure({ adapter: new Adapter() })

describe('ProductFamilyList', () => {
  const mocks = [
    {
      id: '1',
      name: 'bar',
      delimiter: '::',
      familyId: '1'
    },
    {
      id: '2',
      name: 'foosle',
      delimiter: ':',
      familyId: '2'
    }
  ]
  const mockGetAll = jest.fn
  const mockprops = {
    getAll: {
      productFamilyCategory: {
        associatedFamilies: mocks
      }
    }
  }
  test('it at least shallow renders and match snapshot', () => {
    const wrapper = shallow(<ProductFamilyList {...mockprops} />)
    wrapper.setProps(mockprops)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
