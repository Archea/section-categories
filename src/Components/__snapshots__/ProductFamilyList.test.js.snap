// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFamilyList it at least shallow renders and match snapshot 1`] = `
<div
  className="ProductFamilyList"
>
  <h3 />
  <ProductFamily
    key="1"
    productFamily={
      Object {
        "delimiter": "::",
        "familyId": "1",
        "id": "1",
        "name": "bar",
      }
    }
  />
  <ProductFamily
    key="2"
    productFamily={
      Object {
        "delimiter": ":",
        "familyId": "2",
        "id": "2",
        "name": "foosle",
      }
    }
  />
</div>
`;
